src/fsharp/FSharp.Core/control.fs

```diff
@@ -1541,8 +1541,8 @@ namespace Microsoft.FSharp.Control
             let continuation (completedTask : Task<_>) : unit =
                 args.aux.trampolineHolder.Protect((fun () ->
                     if completedTask.IsCanceled then
-                        if useCcontForTaskCancellation then args.aux.ccont(new OperationCanceledException())
-                        else args.aux.econt (ExceptionDispatchInfo.Capture(new TaskCanceledException()))
+                        if useCcontForTaskCancellation then args.aux.ccont(new OperationCanceledException(args.aux.token))
```

Comment:
@eiriktsarpalis @matthid What is this were  ``if args.aux.token.IsCancellationRequested then new OperationCanceledException(args.aux.token) else new OperationCanceledException() ``?   

I suspect a lot of corresponding C# code would do the same thing: if a cancellation continuation needs to be converted into a OCE, then you would attach the current cancellation token in scope to the OCE, at least on the condition that that the current cancellation token in scope is in the cancelled state,